Source/SwiftLintFramework/Rules/Idiomatic/FileNameNoSpaceRule.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+import SourceKittenFramework
+
+public struct FileNameNoSpaceRule: ConfigurationProviderRule, OptInRule {
+    public var configuration = FileNameNoSpaceConfiguration(
+        severity: .warning,
+        excluded: []
+    )
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "file_name_no_space",
+        name: "File Name No Space",
+        description: "File name should not contain any whitespace.",
+        kind: .idiomatic
+    )
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        guard let filePath = file.path,
+            case let fileName = filePath.bridge().lastPathComponent,
+            !configuration.excluded.contains(fileName) else {
+            return []
+        }
+
+        let whitespaceRegex = regex("(?:[\\s])")
```

Comment:
Rather than use a regex, why not use `CharacterSet`? Should be more performant and simpler.

Reply:
Sounds good. 👍. Have never used a CharacterSet, so just defaulted to what I know, regex. Looks like it already has a [whitespace](https://developer.apple.com/documentation/foundation/characterset/1779930-whitespaces) static property so it makes it easy.